lib/graphs.ts

```diff
@@ -1,89 +1,126 @@
+    const difference = Math.abs(
+      targetDate.getTime() - generationData[i].datetime_utc.getTime()
+    );
+    const closestDifference = Math.abs(
+      targetDate.getTime() - generationData[closest].datetime_utc.getTime()
+    );
+    if (difference < closestDifference) {
+      closest = i;
+    }
   }
-  return 0;
-};
 
-export const outputDataOverDateRange = <
-  T extends Pick<ForecastDataPoint, 'target_datetime_utc'>
->(
-  forecastData: T[],
-  start_date: Date,
-  end_date: Date
+  return closest;
+}
+
+export const generationDataOverDateRange = (
+  generationData: GenerationDataPoint[],
+  startDate: Date,
+  endDate: Date
 ) => {
-  const start_index = getClosestForecastIndex(forecastData, start_date);
-  const end_index = getClosestForecastIndex(forecastData, end_date);
-  if (forecastData)
-    forecastData = forecastData.slice(start_index, end_index + 1);
-  return forecastData;
+  const startIndex = getClosestForecastIndex(generationData, startDate);
+  const endIndex = getClosestForecastIndex(generationData, endDate);
+  return generationData.slice(startIndex, endIndex + 1);
 };
 
 export const getGraphStartDate = (currentTime: number) => {
   const currentDate = new Date(currentTime);
   return new Date(
-    currentDate.getFullYear(),
-    currentDate.getMonth(),
-    currentDate.getHours() > 20
-      ? currentDate.getDate() + 1
-      : currentDate.getDate(),
-    8
+    Date.UTC(
+      currentDate.getUTCFullYear(),
+      currentDate.getUTCMonth(),
+      currentDate.getUTCHours() > 20
+        ? currentDate.getUTCDate() + 1
+        : currentDate.getUTCDate(),
+      8
+    )
   );
 };
 
 export const getGraphEndDate = (currentTime: number) => {
   const currentDate = new Date(currentTime);
   return new Date(
-    currentDate.getFullYear(),
-    currentDate.getMonth(),
-    currentDate.getHours() > 20
-      ? currentDate.getDate() + 1
-      : currentDate.getDate(),
-    20
+    Date.UTC(
+      currentDate.getUTCFullYear(),
+      currentDate.getUTCMonth(),
+      currentDate.getUTCHours() > 20
+        ? currentDate.getUTCDate() + 1
+        : currentDate.getUTCDate(),
+      20
+    )
   );
 };
 
 /**
  * @returns the index of the forecasted date that is closest to the current time
  */
-export const getCurrentTimeForecastIndex = (
-  forecast_values: ForecastDataPoint[]
-) => {
-  return getClosestForecastIndex(forecast_values, new Date());
-};
+export function getCurrentTimeGenerationIndex(
+  generationData: GenerationDataPoint[]
+) {
+  return getClosestForecastIndex(generationData, new Date());
+}
+
+export function makeGraphable(generationData: GenerationDataPoint[]) {
+  return generationData.map((point) => ({
+    ...point,
+    datetime_utc: point.datetime_utc.getTime(),
+  }));
+}
+
+export function addTimePoint(
+  generationData: GenerationDataPoint[],
+  date: Date
+) {
+  const generationDataInterpolated = generationData.map((data) => ({
+    ...data,
+  }));
+
+  const forecastValuePeriod = 15;
```

Comment:
Can we infer this from the data? Just need to take the difference in time between the closestTime and the next index. Would help when we implement actuals, which have more frequent data.